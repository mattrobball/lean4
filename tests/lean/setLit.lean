structure Foo where
  val := 10

instance : Singleton α (List α) where
  singleton a := [a]

instance : Insert α (List α) where
  insert a as := a :: as

def f1 : Foo := {}

def f2 : List Nat := {}

def f3 : List Nat := {1, 2}

def f4 (a : Nat) : List Nat := {a}

def f5 (val : Nat) : Foo := {val}

def f6 (val : Nat) : List Nat := {val}

def f7 : String := {}

def f8 (val : Nat) : String := { val }
