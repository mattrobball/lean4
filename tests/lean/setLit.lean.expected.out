setLit.lean:22:19-22:21: error: overloaded, errors 
  failed to synthesize instance
    EmptyCollection String
  
  fields missing: 'data'
setLit.lean:24:31-24:38: error: overloaded, errors 
  failed to synthesize instance
    Singleton Nat String
  
  24:33 'val' is not a field of structure 'String'
